File: packages/rune-android/android/RuneKit/src/main/java/com/rune/kit/runtime/JSRuntimeAdapter.kt
package com.rune.kit.runtime

interface JSRuntimeAdapter {
  var onException: ((JsRuntimeException) -> Unit)?
  fun setGlobalObject(name: String, value: Any)
  fun setGlobalFunction(name: String, fn: (Array<Any?>) -> Any?)
  fun evaluate(code: String)
  fun evaluateAsync(code: String) {
    evaluate(code)
  }
  fun callGlobal(name: String, args: Array<Any?> = emptyArray()): Any?
  fun callGlobalAsync(name: String, args: Array<Any?> = emptyArray()) {
    callGlobal(name, args)
  }
}

File: packages/zynth-ios/ios/ZynthKit/src/runtime/JSRuntimeAdapter.swift
import Foundation

public protocol JSRuntimeAdapter: AnyObject {
  var onException: ((JsRuntimeException) -> Void)? { get set }
  func setGlobalObject(_ name: String, _ value: Any)
  func setGlobalFunction(_ name: String, _ fn: @escaping ([Any]) -> Any?)
  func evaluate(code: String)
  func callGlobal(_ name: String, args: [Any]) -> Any?
}
